From 4eb847eef232c744fa1ff3aefec04f450b8201b0 Mon Sep 17 00:00:00 2001
From: Jie Chen <jie.a.chen@intel.com>
Date: Mon, 17 Feb 2014 14:48:37 +0800
Subject: [PATCH] disable log2

---
 src/gallium/auxiliary/util/u_math.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/gallium/auxiliary/util/u_math.h b/src/gallium/auxiliary/util/u_math.h
index 90b421e..abea2ca 100644
--- a/src/gallium/auxiliary/util/u_math.h
+++ b/src/gallium/auxiliary/util/u_math.h
@@ -138,7 +138,8 @@ roundf(float x)
 
 
 #ifdef PIPE_OS_ANDROID
-
+//ndk has them already
+#if 0
 static INLINE
 double log2(double d)
 {
@@ -155,7 +156,7 @@ float log2f(float f)
 {
    return logf(f) * (float) (1.0 / M_LN2);
 }
-
+#endif
 #endif
 
 
-- 
1.8.3.2

