From 1736358fa9cde87e67179ca5a8c82c8ebf3fd048 Mon Sep 17 00:00:00 2001
From: Jie Chen <jie.a.chen@intel.com>
Date: Mon, 17 Feb 2014 14:45:40 +0800
Subject: [PATCH] suppress warning error in ppapi

---
 ppapi/ppapi_proxy.gypi |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/ppapi/ppapi_proxy.gypi b/ppapi/ppapi_proxy.gypi
index 6821ea4..5ac7e21 100644
--- a/ppapi/ppapi_proxy.gypi
+++ b/ppapi/ppapi_proxy.gypi
@@ -7,6 +7,9 @@
     'variables': {
       'ppapi_proxy_target': 0,
     },
+    'cflags': [
+      '-fpermissive',
+    ],
     'target_conditions': [
       # This part is shared between the targets defined below.
       ['ppapi_proxy_target==1', {
-- 
1.7.9.5

